### PR TITLE
Incorporating Link Semantics

### PR DESCRIPTION
**Pull request by Piinks**
## Description

Incorporates link specific semantics. Dependent on engine change: 
~https://github.com/flutter/engine/pull/12453~ 
~https://github.com/flutter/engine/pull/12972~
~https://github.com/flutter/engine/pull/13001~
https://github.com/flutter/engine/pull/13009

Will require a manual engine roll.

This change also incorporates the link semantics with TextSpans that use TapGestureRecognizers.

## Related Issues

Part of #36306 

## Tests

Updated semantics related tests.
Added test to check that the SemanticsFlag.isLink is added properly to relevant TextSpans

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [ ] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/semantics/semantics.dart [hunk 1]
```diff
@@ -670,6 +671,11 @@ class SemanticsProperties extends DiagnosticableTree {
   /// is focused.
   final bool button;
 
+  /// If non-null, indicates that this subtree represents a link.
+  ///
+  /// TalkBack/VoiceOver provides users with a hint when a link is focused.
```
Comment (goderbauer) [inline packages/flutter/lib/src/semantics/semantics.dart]: Document that talkback just identifies this as a button.

File: packages/flutter/lib/src/rendering/paragraph.dart [hunk 2]
```diff
@@ -861,6 +861,7 @@ class RenderParagraph extends RenderBox
           if (info.recognizer is TapGestureRecognizer) {
             final TapGestureRecognizer recognizer = info.recognizer;
             configuration.onTap = recognizer.onTap;
+            configuration.isLink = true;
```
Comment (goderbauer) [inline packages/flutter/lib/src/rendering/paragraph.dart]: Add a test for this?

**Review by goderbauer — APPROVED** [review]
LGTM with the two nits.

File: packages/flutter/lib/src/rendering/paragraph.dart (hunk 2 above)
Comment (Piinks) [reply packages/flutter/lib/src/rendering/paragraph.dart]: Done

**Comment by Piinks** [comment]
Will need to update the engine version after https://github.com/flutter/engine/pull/12972 lands

**Comment by Piinks** [comment]
Landing on red to fix the build. :D

**Pull request merged**